### PR TITLE
Add equality testing for Text objects

### PR DESCRIPTION
**Pull request by malmaud**

### CODE REVIEW

**Comment by TotalVerb** [comment]
Needs a corresponding `hash(::Text)` definition.

**Comment by malmaud** [comment]
I am really unclear why the new `hash` test would fail only on i686

File: base/docs/utils.jl [hunk 1]
```diff
@@ -72,6 +72,9 @@ print(io::IO, t::Text) = print(io, t.content)
 print{F <: Function}(io::IO, t::Text{F}) = t.content(io)
 show(io::IO, t::Text) = print(io, t)
 
+=={T<:Union{HTML, Text}}(t1::T, t2::T) = t1.content == t2.content
+hash{T<:Union{HTML, Text}}(t::T, h::UInt64) = hash(t.content, h)
```
Comment (tkelman) [inline base/docs/utils.jl]: UInt ?

Comment (malmaud) [inline base/docs/utils.jl]: Ah right

File: base/docs/utils.jl [hunk 2]
```diff
@@ -72,6 +72,9 @@ print(io::IO, t::Text) = print(io, t.content)
 print{F <: Function}(io::IO, t::Text{F}) = t.content(io)
 show(io::IO, t::Text) = print(io, t)
 
+=={T<:Union{HTML, Text}}(t1::T, t2::T) = t1.content == t2.content
+hash{T<:Union{HTML, Text}}(t::T, h::UInt) = hash(t.content, h)
```
Comment (StefanKarpinski) [inline base/docs/utils.jl]: Probably want to mix in a static hash of the type here unless you want cross-type values to hash the same (e.g. `hash(HTML("foo")) == hash(Text("foo")) == hash("foo")`. As an aside, we really need to make this easier to do correctly.

Comment (malmaud) [inline base/docs/utils.jl]: Ah, good catch.

**Pull request merged**